08_files_lab/01.py

```diff
@@ -1,34 +1,28 @@
-"""
-Write a program that takes 2 file names
-and appends the second file's contents to
-the end of the first one.
-
-So if file 'a.txt' has:
-
-a1
-a2
-
-and file 'b.txt' has
-
-b1
-b2
-b3
-
-then after running
-   python 01.py a.txt b.txt
-a.txt will have
-
-a1
-a2
-b1
-b2
-b3
-
-and b.txt will have
-
-b1
-b2
-b3
-
-"""
+#!/usr/bin/python
+import sys, os
+
+if len(sys.argv) == 1:
+	sys.exit('usage %s <text file1> <text file2(optinal)>' % sys.argv[0])
+
+if len(sys.argv) == 2:
+	with open(sys.argv[1]) as f1:
+		Text1 = f1
+		with open('newFile.txt', 'w') as f2:
+			for line in Text1:
+				f2.write(line)
+	sys.exit()			
+elif len(sys.argv) == 3:
+	with open(sys.argv[1]) as f1:
+		Text1 = f1
+		with open(sys.argv[2], 'a') as f2:
+			for line in Text1:
+				f2.write(line)
+elif len(sys.argv) == 4:
+	with open(sys.argv[1]) as f1:
+		with open(sys.argv[2]) as f2:
+			with open(sys.argv[3], 'a') as f3:
+				for line in f1:
+					f3.write(line)
+				for line in f2:
+					f3.write(line)
```

Comment:
I think you have too many ifs there- the program should receive 2 file names and append the second one to the end of the first one.

One easy way is:

```
import sys

with open(sys.argv[1], 'a') as target:
    with open(sys.argv[2]) as src:
        target.write(src.read())
```

(And don't worry about the usage block just yet, we'll return to that when we'll learn about argparse and exceptions)

The problem with the suggested solution is it doesn't work well on large files, because it reads the entire file into memory. The following is a slight improvement though a bit longer:

```
import sys

CHUNK_SIZE = 1024

with open(sys.argv[1], 'a') as target:
    with open(sys.argv[2]) as src:
        while True:
            chunk = src.read(CHUNK_SIZE)
            if chunk == '':
                break
            target.write(chunk)
```